src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -762,31 +794,280 @@ public void Dispose()
+            /// object may be missing some information, specifically in the instances where PingReply objects are
+            /// utilised to perform a traceroute.
+            /// </summary>
+            /// <param name="source">The source machine name or IP of the ping.</param>
+            /// <param name="destination">The destination machine name of the ping.</param>
+            /// <param name="reply">The response from the ping attempt.</param>
+            /// <param name="latency">The latency of the ping.</param>
+            /// <param name="bufferSize">The buffer size.</param>
+            /// <param name="pingNum">The sequence number in the sequence of pings to the hop point.</param>
+            internal PingStatus(
+                string source,
+                string destination,
+                PingReply reply,
+                long latency,
+                int bufferSize,
+                uint pingNum)
+                : this(source, destination, reply, pingNum)
+            {
+                _bufferSize = bufferSize;
+                _latency = latency;
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="PingStatus"/> class.
+            /// </summary>
+            /// <param name="source">The source machine name or IP of the ping.</param>
+            /// <param name="destination">The destination machine name of the ping.</param>
+            /// <param name="reply">The response from the ping attempt.</param>
+            /// <param name="pingNum">The sequence number of the ping in the sequence of pings to the target.</param>
+            internal PingStatus(string source, string destination, PingReply reply, uint pingNum)
+            {
+                Ping = pingNum;
+                Reply = reply;
+                Source = source;
+                Destination = destination;
+            }
+
+            // These values can be set manually to skirt issues with the Ping API on Unix platforms
+            // so that we can return meaningful known data that is discarded by the API.
+            private readonly int _bufferSize = -1;
+
+            private readonly long _latency = -1;
+
+            /// <summary>
+            /// Gets the sequence number of this ping in the sequence of pings to the <see cref="Destination"/>
+            /// </summary>
+            public uint Ping { get; }
+
+            /// <summary>
+            /// Gets the source from which the ping was sent.
+            /// </summary>
+            public string Source { get; }
+
+            /// <summary>
+            /// Gets the destination which was pinged.
+            /// </summary>
+            public string Destination { get; }
+
+            /// <summary>
+            /// Gets the target address of the ping.
+            /// </summary>
+            public IPAddress? Address { get => Reply.Status == IPStatus.Success ? Reply.Address : null; }
```

Comment:
Interesting. Querying the Reply.Address in the latest build I got from CI appears to be... broken. 
```powershell

 PS>  $a.Reply.Address

OverloadDefinitions
-------------------
System.Object&, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e Address(int )

```

I'll have to make sure I rebase this PR and see if this is resolved in a later .NET core preview. Querying the address in different ways does work, but something is a bit screwy with the auto-property. Gonna look into this probably this evening, should hopefully not be a complex fix. 🤞

Reply:
FIgured it out. Pushing a fix.